src/math/log.c

```diff
@@ -0,0 +1,115 @@
+	    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+	    1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+	    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+	    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+	    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+	    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+	uint64_t v;
+	uint32_t x;
+	int num_left_shifts;
+	int shift_factor;
+	unsigned int slice_temp;
+	/*  Skipping for-loop */
+	/*  Unroll the loop so there will be no branching. */
+	/*  Loop unroll execute 20% faster (CPU cycle count )than for loop */
+	/*  Every iteration ,look for leading zeros are in the high */
+	/*  byte of V, and shift them out to the left. Continue with the */
+	/*  shifted V for as many bytes as it has. */
+	/*  The index is the high byte of the input plus 1 to make it a */
+	/*  one-based index. */
+	/*  Index into the number-of-leading-zeros lookup table.  This lookup */
+	/*  table takes in a byte and returns the number of leading zeros in the */
+	/*  binary representation. */
+	shift_factor = iv[u >> 24];
+	/*  Left-shift out all the leading zeros in the high byte. */
+	v = (uint64_t)u << shift_factor;
+	/*  Update the total number of left-shifts */
+	num_left_shifts = shift_factor;
+	shift_factor = iv[v >> 24];
+	/*  Left-shift out all the leading zeros in the high byte. */
+	v <<= shift_factor;
+	/*  Update the total number of left-shifts */
+	num_left_shifts += shift_factor;
+	shift_factor = iv[v >> 24];
+	/*  Left-shift out all the leading zeros in the high byte. */
+	v <<= shift_factor;
+	/*  Update the total number of left-shifts */
+	num_left_shifts += shift_factor;
+	shift_factor = iv[v >> 24];
+	/*  Left-shift out all the leading zeros in the high byte. */
+	/*  Update the total number of left-shifts */
+	num_left_shifts += shift_factor;
+	/*  The input has been left-shifted so the most-significant-bit is a 1. */
+	/*  Reinterpret the output as unsigned with one integer bit, so */
+	/*  that 1 <= x < 2. */
+	x = (uint32_t)v << shift_factor;
+	/*  Let Q = int(u).  Then u = Q*2^(-u_fraction_length), */
+	/*  and x = Q*2^num_left_shifts * 2^(1-word_length).  Therefore, */
+	/*  u = x*2^n, where n is defined as: */
+	/*  Extract the high byte of x */
+	/*  Convert the high byte into an index for lookup table */
+	slice_temp = ((v << shift_factor) >> 24);
+	/*  Interpolate between points. */
+	/*  The upper byte was used for the index into lookup table */
+	/*  The remaining bits make up the fraction between points. */
+	return (((BASE2LOG_WORDLENGTH - num_left_shifts) << 16) +
+		iv1[(BASE2LOG_WRAP_SCHAR_BITS & ((x >> 24) + 129)) - 1]) +
+		((((x & BASE2LOG_UPPERBYTES) *
+			   (int64_t)(
+			       iv1[(BASE2LOG_WRAP_SCHAR_BITS & (slice_temp + 130)) - 1] -
```

Comment:
this is still too messy, can you move these into their own lines before the return?